File: Subclass/SearchFieldforLists.m
//
//  SearchFieldforLists.m
//  BeFit
//
//  Created by Jon Brown on 11/30/13.
//
//

#import "SearchFieldforLists.h"

@implementation SearchFieldforLists

-(void)awakeFromNib
{
    
    NSMenu *cellMenu = [[NSMenu alloc] initWithTitle:@"Filter Menu"];
    
    NSMenuItem *item;
    
    item = [[NSMenuItem alloc] initWithTitle:@"Recents"
             
                                       action:NULL
             
                                keyEquivalent:@""];
    
    [item setTag:NSSearchFieldRecentsMenuItemTag];
    
    [cellMenu insertItem:item atIndex:0];
    
    id searchCell = [self cell];
    
    [searchCell setSearchMenuTemplate:cellMenu];
    
}

@end

File: Subclass/SearchFieldforLists.swift
//
//  SearchFieldforLists.swift
//  BeFit
//
//  Created by Andrej Slegl on 11/6/19.
//

import Foundation
import Cocoa

class SearchFieldforLists: NSSearchField {
    override func awakeFromNib() {
        super.awakeFromNib()
        
        let cellMenu = NSMenu(title: "Filter Menu")
        let item = NSMenuItem(title: "Recents", action: nil, keyEquivalent: "")
        
        item.tag = NSSearchField.recentsMenuItemTag
        cellMenu.insertItem(item, at: 0)
        
        (cell as? NSSearchFieldCell)?.searchMenuTemplate = cellMenu
    }
}
